File: app/models/story.rb
class Story < ActiveRecord::Base
  attr_accessible :preferences, :source, :team_member, :source_uid

  belongs_to :team_member
  belongs_to :source

  delegate :source_type, to: :source

  serialize :preferences, ActiveRecord::Coders::Hstore

  validates_uniqueness_of :source_uid

  pusherable('dashboard')
end

File: app/assets/javascripts/models/story.js
App.Story = DS.Model.extend({
  source: DS.belongsTo('App.Source'),
  teamMember: DS.belongsTo('App.TeamMember'),
  preferences: DS.attr('object'),
  sourceUid: DS.attr('string'),
  createdAt: DS.attr('date'),
  updatedAt: DS.attr('date')
});
